subprojects/docs/src/docs/userguide/jvm/java_plugin.adoc

```diff
@@ -546,6 +546,8 @@ include::{snippetsPath}/java/incrementalAnnotationProcessing/groovy/processor/sr
 
 Both categories have the following limitations:
 
+* They have to read `CLASS` or `RUNTIME` retention annotations only.
+    Reading `SOURCE` retention annotations is possible for "isolating" processors on their own. But it will produce incorrect results when a different "aggregating" processor process the same types as the "isolating" processor, so it should not be used.
```

Comment:
This sentence just weakens the point above and could lead to confusion, I'd remove it completely.

Reply:
Removed explanations that might confuse users.